presto-phoenix/src/main/java/io/prestosql/plugin/phoenix/PhoenixMetadata.java

```diff
@@ -0,0 +1,455 @@
+                String columnName = column.getName();
+                if (uppercase) {
+                    columnName = columnName.toUpperCase(ENGLISH);
+                }
+                columnNames.add(columnName);
+                columnTypes.add(column.getType());
+                String typeStatement = phoenixClient.toWriteMapping(session, column.getType()).getDataType();
+                if (rowkeyColumns.contains(column)) {
+                    typeStatement += " not null";
+                    pkNames.add(columnName);
+                }
+                columnList.add(format("%s %s", columnName, typeStatement));
+            }
+
+            ImmutableList.Builder<String> tableOptions = ImmutableList.builder();
+            PhoenixTableProperties.getSaltBuckets(tableProperties).ifPresent(value -> tableOptions.add(TableProperty.SALT_BUCKETS + "=" + value));
+            PhoenixTableProperties.getSplitOn(tableProperties).ifPresent(value -> tableOptions.add("SPLIT ON (" + value.replace('"', '\'') + ")"));
+            PhoenixTableProperties.getDisableWal(tableProperties).ifPresent(value -> tableOptions.add(TableProperty.DISABLE_WAL + "=" + value));
+            PhoenixTableProperties.getDefaultColumnFamily(tableProperties).ifPresent(value -> tableOptions.add(TableProperty.DEFAULT_COLUMN_FAMILY + "=" + value));
+            PhoenixTableProperties.getBloomfilter(tableProperties).ifPresent(value -> tableOptions.add(HColumnDescriptor.BLOOMFILTER + "='" + value + "'"));
+            PhoenixTableProperties.getVersions(tableProperties).ifPresent(value -> tableOptions.add(HConstants.VERSIONS + "=" + value));
+            PhoenixTableProperties.getMinVersions(tableProperties).ifPresent(value -> tableOptions.add(HColumnDescriptor.MIN_VERSIONS + "=" + value));
+            PhoenixTableProperties.getCompression(tableProperties).ifPresent(value -> tableOptions.add(HColumnDescriptor.COMPRESSION + "='" + value + "'"));
+            PhoenixTableProperties.getTimeToLive(tableProperties).ifPresent(value -> tableOptions.add(HColumnDescriptor.TTL + "=" + value));
+
+            String sql = format(
+                    "CREATE %s TABLE %s (%s , CONSTRAINT PK PRIMARY KEY (%s)) %s",
+                    immutable,
+                    getEscapedTableName(schema, table),
+                    join(", ", columnList.build()),
+                    join(", ", pkNames.build()),
+                    join(", ", tableOptions.build()));
+
+            phoenixClient.execute(session, sql);
+
+            return new PhoenixOutputTableHandle(
+                    Optional.of(""),
+                    schema,
+                    table,
+                    columnNames.build(),
+                    columnTypes.build(),
+                    "",
+                    hasUUIDRowkey);
+        }
+        catch (SQLException e) {
+            throw new PrestoException(PHOENIX_METADATA_ERROR, "Error creating Phoenix table", e);
```

Comment:
As mentioned above, we should throw `ALREADY_EXISTS` if possible when the table already exists. Maybe Phoenix has an error code or well-known error message to compare against?

Throwing `ALREADY_EXISTS` here will allow `CREATE TABLE IF NOT EXISTS` to work correctly if someone else created the table after the engine already checked.